test/test_common/network_utility.cc

```diff
@@ -0,0 +1,33 @@
+#include "network_utility.h"
+
+#include "common/runtime/runtime_impl.h"
+
+#include <mutex>
+
+namespace Network {
+namespace Test {
+namespace {
+const int kFirstEphemeralPort = 40152;
+const int kLastEphemeralPort = 65535;
+
+int selectFirstPort() {
+  uint64_t range = kLastEphemeralPort - kFirstEphemeralPort + 1;
+  Runtime::RandomGeneratorImpl rng;
```

Comment:
I'm not crazy about this either, but I'm OK with it for now if we are going to replace it with something better soonish. A possible better thing to do might be to just not do the random at all for this and just let it cycle through in order.